`make build-images` fails with no REGISTRY variable set

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via 
-->

**What happened**:

if `REGISTRY` is not as an environment arg or passed to the Makefile target when running `make build-images` it fails.

```shell
❯ make build-images
mkdir -p ~/.kube/
mkdir -p /home/markmandel/workspace/agones/build//.gocache
mkdir -p /home/markmandel/workspace/agones/build//.config/gcloud
mkdir -p ~/.config/helm
mkdir -p ~/.cache/helm
docker run --privileged --rm tonistiigi/binfmt:qemu-v6.2.0 --install arm64
installing: arm64 qemu-aarch64 already registered
{
  "supported": [
    "linux/amd64",
    "linux/arm64",
    "linux/386"
  ],
  "emulators": [
    "python3.11",
    "qemu-aarch64"
  ]
}
make ensure-image IMAGE_TAG=agones-build:5bab735929 BUILD_TARGET=build-build-image
make[1]: Entering directory '/home/markmandel/workspace/agones/build'
make[1]: Leaving directory '/home/markmandel/workspace/agones/build'
docker run --rm -e "CGO_ENABLED=0" -e "GO111MODULE=on" -w /go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /home/markmandel/workspace/agones:/go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.gomod:/go/pkg/mod -v /home/markmandel/workspace/agones/build//.gocache:/root/.cache/go-build agones-build:5bab735929 go build -mod=vendor -buildvcs=false \
-tags none -o /go/src/agones.dev/agones/cmd/controller/bin/controller.linux.amd64 \
 -ldflags "-X agones.dev/agones/pkg.Version=1.41.0-dev-2abfd71" -installsuffix cgo agones.dev/agones/cmd/controller
docker run --rm -e "CGO_ENABLED=0" -e "GOOS=linux" -e "GOARCH=arm64" -e "GO111MODULE=on" -w /go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /home/markmandel/workspace/agones:/go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.gomod:/go/pkg/mod -v /home/markmandel/workspace/agones/build//.gocache:/root/.cache/go-build agones-build:5bab735929 go build -mod=vendor -buildvcs=false \
-tags none -o /go/src/agones.dev/agones/cmd/controller/bin/controller.linux.arm64 \
 -ldflags "-X agones.dev/agones/pkg.Version=1.41.0-dev-2abfd71" -installsuffix cgo agones.dev/agones/cmd/controller
docker run --rm -v /home/markmandel/workspace/agones/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /home/markmandel/workspace/agones:/go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.gomod:/go/pkg/mod -v /home/markmandel/workspace/agones/build//.gocache:/root/.cache/go-build agones-build:5bab735929 /go/src/agones.dev/agones/build/extract-licenses.sh
docker run --rm -v /home/markmandel/workspace/agones/build//.config/gcloud:/root/.config/gcloud -v ~/.kube/:/root/.kube -v ~/.config/helm:/root/.config/helm -v ~/.cache/helm:/root/.cache/helm -v /home/markmandel/workspace/agones:/go/src/agones.dev/agones -v /home/markmandel/workspace/agones/build//.gomod:/go/pkg/mod -v /home/markmandel/workspace/agones/build//.gocache:/root/.cache/go-build agones-build:5bab735929 /go/src/agones.dev/agones/build/build-required-src-dist.sh
docker build /home/markmandel/workspace/agones/cmd/controller/ --tag=/agones-controller:1.41.0-dev-2abfd71-amd64
[+] Building 0.0s (0/0)                                                                                                                                                                                                                                                                                                                                docker:default
ERROR: invalid tag "/agones-controller:1.41.0-dev-2abfd71-amd64": invalid reference format
make: *** [Makefile:528: build-controller-image-amd64] Error 1
```

**What you expected to happen**:

I'd expect it to build, but produce an image that has an invalid registry, such as `local` so that there is nowhere for it to be pushed unless the end user does actually specify a `REGISTRY` somewhere.

**How to reproduce it (as minimally and precisely as possible)**:

Make sure `REGISTRY` is not set anywhere (I set mind a lot in the `local-includes` folder), and run `make build-images`

**Anything else we need to know?**:

Let's make sure to set the registry to something that's not a valid registry 😃 that way it can't be pushed anywhere (so **definitely not** `agones-images`.

**Environment**:
- Agones version:  dev
- Kubernetes version (use `kubectl version`): N/A
- Cloud provider or hardware configuration: N/A
- Install method (yaml/helm): N/A
- Troubleshooting guide log(s): See above
- Others: N/A

--- a/build/Makefile
+++ b/build/Makefile
@@ -99,6 +99,7 @@ common_mounts = -v $(build_path)/.config/gcloud:/root/.config/gcloud \
 				-v $(build_path)/.gomod:/go/pkg/mod
 
 # Its possible for this to change in the future, so it gets its own variable.
+REGISTRY ?= local
 workdir_path = $(mount_path)
 
 build_version = $(call sha,$(build_path)/build-image/Dockerfile)

